### PR TITLE
Add otc_coalescing_strategy config option

### PR DESCRIPTION
**Pull request by cooldoger**

### CODE REVIEW

File: cassandra-scheduler/src/dist/conf/scheduler.yml [hunk 1]
```diff
@@ -139,6 +139,7 @@ cassandra:
      streaming_socket_timeout_in_ms : ${CASSANDRA_STREAMING_SOCKET_TIMOUT_IN_MS:-86400000}
      enable_scripted_user_defined_functions : ${CASSANDRA_ENABLE_SCRIPTED_USER_DEFINED_FUNCTIONS:-false}
      max_value_size_in_mb : ${CASSANDRA_MAX_VALUE_SIZE_IN_MB:-256}
+     otc_coalescing_strategy : ${OTC_COALESCING_STRATEGY:-"TIMEHORIZON"}
```
Comment (verma7) [inline cassandra-scheduler/src/dist/conf/scheduler.yml]: Please prefix the env var with `CASSANDRA_` for consistency.

File: cassandra-scheduler/src/test/resources/scheduler.yml [hunk 2]
```diff
@@ -139,6 +139,7 @@ cassandra:
      streaming_socket_timeout_in_ms : ${CASSANDRA_STREAMING_SOCKET_TIMOUT_IN_MS:-86400000}
      enable_scripted_user_defined_functions : ${CASSANDRA_ENABLE_SCRIPTED_USER_DEFINED_FUNCTIONS:-false}
      max_value_size_in_mb : ${CASSANDRA_MAX_VALUE_SIZE_IN_MB:-256}
+     otc_coalescing_strategy : ${OTC_COALESCING_STRATEGY:-"DISABLED"}
```
Comment (verma7) [inline cassandra-scheduler/src/test/resources/scheduler.yml]: Same here. Also, can you set the same default value everywhere, specifically the one that ships as default if this var is not set.

File: cassandra-scheduler/src/test/resources/update-scheduler.yml [hunk 3]
```diff
@@ -138,6 +138,7 @@ cassandra:
      streaming_socket_timeout_in_ms : ${CASSANDRA_STREAMING_SOCKET_TIMOUT_IN_MS:-86400000}
      enable_scripted_user_defined_functions : ${CASSANDRA_ENABLE_SCRIPTED_USER_DEFINED_FUNCTIONS:-false}
      max_value_size_in_mb : ${CASSANDRA_MAX_VALUE_SIZE_IN_MB:-256}
+     otc_coalescing_strategy : ${OTC_COALESCING_STRATEGY:-"DISABLED"}
```
Comment (verma7) [inline cassandra-scheduler/src/test/resources/update-scheduler.yml]: Same here.

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

File: cassandra-scheduler/src/dist/conf/scheduler.yml [hunk 4]
```diff
@@ -41,7 +41,7 @@ cassandra:
      incremental_backups: false
      auto_snapshot: false
      snapshot_before_compaction: false
-     enable_user_defined_functions: false
+     enable_user_defined_functions: ${CASSANDRA_ENABLE_USER_DEFINED_FUNCTIONS:-false}
```
Comment (mohitsoni) [inline cassandra-scheduler/src/dist/conf/scheduler.yml]: Can you please move this to a separate PR ? As this doesn't relate to coalescing strategy.

Comment (cooldoger) [inline cassandra-scheduler/src/dist/conf/scheduler.yml]: Removed. Will create separate PR for this.

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

**Comment by cooldoger** [comment]
Rebased. Please merge.

**Comment by cooldoger** [comment]
@nickbp Seems like Jenkins server has problem. It was passed for this PR. Please check.

Also coalescing is an important option to expose, as default setting has performance regression [CASSANDRA-12676](https://issues.apache.org/jira/browse/CASSANDRA-12676), customers should try to set it to `disabled`.

**Comment by NimaVaziri** [comment]
retest this please (looks like there was a network error)

**Comment by cooldoger** [comment]
@NimaVaziri @nickbp tests are passed, please merge.
